sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ConditionalStructureSameImplementation.cs

```diff
@@ -67,42 +73,50 @@ protected sealed override void Initialize(SonarAnalysisContext context)
             context.RegisterSyntaxNodeActionInNonGenerated(
                 c =>
                 {
-                    var switchSection = (SwitchSectionSyntax) c.Node;
+                    var switchSection = (SwitchSectionSyntax)c.Node;
+
+                    if (GetStatements(switchSection).Where(IsNotIgnored).AreLessThan(2))
+                    {
+                        return;
+                    }
+
                     var precedingSection = switchSection
                         .GetPrecedingSections()
                         .FirstOrDefault(
                             preceding => EquivalenceChecker.AreEquivalent(switchSection.Statements, preceding.Statements));
 
                     if (precedingSection != null)
                     {
-                        ReportSection(c, switchSection, precedingSection);
+                        ReportSyntaxNode(c, switchSection, precedingSection, "case");
                     }
                 },
                 SyntaxKind.SwitchSection);
         }
 
-        private static void CheckStatement(SyntaxNodeAnalysisContext context, StatementSyntax statementToCheck,
+        private static IEnumerable<StatementSyntax> GetStatements(SwitchSectionSyntax switchSection)
+        {
+            return Enumerable.Empty<StatementSyntax>()

```

Comment:
Do you really need the first `Enumerable.Empty<>().Union()`? Can't you start directly with the content?